pkg/simpleschema/transform_test.go

```diff
@@ -360,6 +360,103 @@ func TestBuildOpenAPISchema(t *testing.T) {
 			want:    nil,
 			wantErr: true,
 		},
+		{
+			name: "Schema with CEL validation - immutable filed",
+			obj: map[string]interface{}{
+				"immutableFoo": "string | validation={\"rule\": \"self == oldSelf\", \"message\": \"can't be changed\"}",
```

Comment:
I was thinking that maybe we can also have a simpler support for validation, like name: `string | validation="self.name != 'something'"`. And honestly i'm not really convinced that smashing jsons in the simple schema is a good thing, idk just sharing my thoughts :) - let's keep for now and see what other folks think.